src/renderer/redux/actions/publish.js

```diff
@@ -107,13 +108,18 @@ export const doPublish = (params: PublishParams): ThunkAction => {
   }
 
   const publishPayload = {
-    file_path: filePath,
     name,
     channel_name: channelName,
     bid,
     metadata,
   };
 
+  if (filePath) {
+    publishPayload.file_path = filePath;
+  } else {
+    publishPayload.sources = source;
```

Comment:
not trying to make dumb comments, but maybe this should be named `sources` and not `source`